src/Compilers/CSharp/Desktop/CommandLine/CommandLineParser.cs

```diff
@@ -554,8 +554,16 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                                 {
                                     generalDiagnosticOption = ReportDiagnostic.Error;
 
-                                    // Clear specific warnaserror options (since last /warnaserror flag on the command line always wins).
+                                    // Reset specific warnaserror options (since last /warnaserror flag on the command line always wins),
+                                    // and bump warnings to errors.
                                     warnAsErrors.Clear();
+                                    foreach (var key in diagnosticOptions.Keys)
+                                    {
+                                        if (diagnosticOptions[key] == ReportDiagnostic.Warn)
```

Comment:
I see.. So we want to make /warnaserror:id only apply to  warnings unlike before where it would apply to infos as well. However /nowarn and /nowarn:id would continue to suppress infos like before. Correct? I thought we said we would allow /warnaserror:id to promote infos to error yesterday but maybe I am misremembering.